billboardd/daemon.h

```diff
@@ -282,7 +288,9 @@ class Daemon : public QObject {
         }
 
         void contextChanged() {
-            updateTimer.start();
+            if (tklock.isLocked()) {
```

Comment:
There's a more subtle issue - if Billboard is started in locked state (see other comment), then `isLocked()` will never become true until the screen is unlocked and locked again. So please remove it (here and the function altogether) and let's just use the `locked()` signal for doing an additional `contextChanged()`.

Also, when the screen isn't locked, updating the Billboard isn't all that costly, as the screen is on and the CPU is probably running, as the user is interacting with the app.